### PR TITLE
docs: Update DEVELOPERS.md

### PR DESCRIPTION
**Pull request by flemzord**

### CODE REVIEW

File: DEVELOPERS.md [hunk 1]
```diff
@@ -1,18 +1,53 @@
...
+earthly -P github.com/formancehq/earthly/bootstrap/+deploy-base-component --component nats
+earthly -P github.com/formancehq/earthly/bootstrap/+deploy-base-component --component jaeger
+earthly -P github.com/formancehq/earthly/bootstrap/+deploy-base-component --component goproxy
+earthly -P github.com/formancehq/earthly/bootstrap/+deploy-base-component --component ghcr-registry
+earthly -P github.com/formancehq/earthly/bootstrap/+deploy-base-component --component buildkitd
 ```
-moon run openapi:build
+You can now start deploying all components.
+```bash
+earthly -P +deploy-all
```
Comment (paul-nicolas) [inline DEVELOPERS.md:38]: missing --push am I right @gfyrag ?

Comment (paul-nicolas) [reply DEVELOPERS.md:38]: For all earthly commands

Comment (gfyrag) [reply DEVELOPERS.md:38]: indeed

**Review by paul-nicolas — APPROVED** [review]

File: DEVELOPERS.md [hunk 2]
```diff
@@ -1,18 +1,53 @@
...
+    - GORELEASER_KEY: Github token for releases
+- vCluster config:
+    - FORMANCE_DEV_KUBE_TOKEN: Kubernetes token for dev cluster access
+    - FORMANCE_DEV_KUBE_API_SERVER_ADDRESS: dev cluster address
+    - FORMANCE_DEV_TLD: dev domain to access the stack
+    - FORMANCE_DEV_GHCR_REGISTRY: Registry for images
+    - FORMANCE_DEV_USER: User for images
+- Admin config: only for our DNS configuration with Route53
+    - FORMANCE_AWS_KEY_ID: AWS Key ID
+    - FORMANCE_AWS_SECRET_KEY: AWS Secret Key
```
Comment (lwagner-getmomo) [inline DEVELOPERS.md:21]: I assume this is not needed for running locally?

File: DEVELOPERS.md [hunk 3]
```diff
@@ -1,18 +1,53 @@
...
+earthly -P --push github.com/formancehq/earthly/bootstrap/+deploy-base-component --component nats
+earthly -P --push github.com/formancehq/earthly/bootstrap/+deploy-base-component --component jaeger
+earthly -P --push github.com/formancehq/earthly/bootstrap/+deploy-base-component --component goproxy
+earthly -P --push github.com/formancehq/earthly/bootstrap/+deploy-base-component --component ghcr-registry
+earthly -P --push github.com/formancehq/earthly/bootstrap/+deploy-base-component --component buildkitd
 ```
-moon run openapi:build
+You can now start deploying all components.
+```bash
+earthly -P --push +deploy-all
```
Comment (lwagner-getmomo) [inline DEVELOPERS.md:38]: This command tries to push images to `<FORMANCE_DEV_GHCR_REGISTRY>/formancehq/<component>:<tag>`. I don't think that works from outside the formancehq org on GitHub.

File: DEVELOPERS.md [hunk 4]
```diff
@@ -1,18 +1,53 @@
...
-
-## Generate SDK with Moon and Speakeasy
+## How to start developing
+To use Earthly and dev environments in Kubernetes, you need the following environment variables (in .direnv/.env):
+Environment variables are split into several pieces:
+- Project config: License of the various tools used in the repository
+    - SPEAKEASY_API_KEY: License for [Speakeasy](https://www.speakeasyapi.dev/)
+    - GITHUB_TOKEN: Github token for actions
+    - GORELEASER_KEY: Github token for releases
+- vCluster config:
```
Comment (lwagner-getmomo) [inline DEVELOPERS.md:13]: If vCluster is an integral part of the deployment, I feel like this should be expressed explicitly with some link to a documentation for setting this up. As far as I can tell from the vCluster docs, one could just as well just deploy to a barebones kubernetes cluster.

**Review by lwagner-getmomo — COMMENTED** [review]
A few things are a bit unclear. The image deployment is a real blocker for me at the moment.

**Pull request merged**